### PR TITLE
Finalize structure crate

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Finishes the `experiment-structure` crate

## 🔗 Related links

Part of [separation of crates](https://app.asana.com/0/1201707629991362/1201481007343142) (_internal_)

- [Asana task](https://app.asana.com/0/0/1202099498122332/f) _(internal)_

## 🚫  Blocked by

- https://github.com/hashintel/hash/pull/689

## 🔍 What does this change?

- Move `SchemaConfig` to `stateful`: 71428500d0da269992208527f29f8cba5d750a44
- Move the simulation configs to `simulation_structure`: 29fdc2919e85906868199c3c3c6a88ce36c8e531
- Replace `Store` with `Option<(State, Context)>`: a212f8e4d7212b5cb769e1ccc7f7ce48a9c99abb
- Move `ActiveTask` to `execution`: e683bae6dd807474be994b63c50cff589484e489
- Split `comms` from `PackageComms` and remove the `Comms` trait: 092856c373127a9d8e2d9e63d13fb53d97105da0
- Remove generics from all package creators: 9688ee6173ac3c6668a1c80a9a249c97ca0dbca9
- Remove structs, which are not used anymore: 4712c78470d9b505dece22a2686ea7d7f2482670
- Merge `Experiment` into `ExperimentRun`: 7c86c8a8818541ac4f83341181e41cf3181329fe
- Rename `Simulation` to `SimulationSource`: 49831bee68fe09908ffcb9210d6fc052e197e789
- Make conversion between `ExperimentId` to `Uuid` explicit: 8ab338c5b1e8de83a41ceb2bdf2be1cf8ca5426c - This was mainly done because somehow `cargo doc` fails to see the conversion. `cargo rustdoc` works fine.
- Rename `simulation_structure` to `experiment-structure`: b7238f8f552d5196584706cdced9e88c789e3918

## 🐾 Next steps

Move remaining things from `hash_engine_lib` to `experiment-control` and/or `simulation-control`

## 🛡 What tests cover this?

- No logic changed, the CI will pick up the same tests as before

### CODE REVIEW

File: packages/engine/lib/execution/src/package/simulation/context/creator.rs [hunk 1]
```diff
@@ -16,27 +16,29 @@ pub struct ContextPackageCreators {
...
-        creators.insert(
-            ContextPackageName::ApiRequests,
-            Box::new(ApiRequestsCreator),
-        );
-        creators.insert(ContextPackageName::Neighbors, Box::new(NeighborsCreator));
-        Ok(Self { creators })
+    pub fn initialize_for_experiment_run(_config: &PackageInitConfig) -> Result<&'static Self> {
+        static PACKAGE_CREATORS: SyncOnceCell<ContextPackageCreators> = SyncOnceCell::new();
+        PACKAGE_CREATORS.get_or_try_init(|| {
+            tracing::debug!("Contextializing Context Package Creators");
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/package/simulation/context/creator.rs:22]: I can't say I understand this 😂  (not just making a joke about the typo but I don't know what you mean by Contextualising the Package Creators)

File: packages/engine/lib/execution/src/package/simulation/output/creator.rs [hunk 2]
```diff
@@ -16,20 +16,22 @@ pub struct OutputPackageCreators {
...
-        tracing::debug!("Initializing Output Package Creators");
-
-        let mut creators = HashMap::<_, Box<dyn OutputPackageCreator>>::with_capacity(2);
-        creators.insert(OutputPackageName::Analysis, Box::new(AnalysisCreator));
-        creators.insert(OutputPackageName::JsonState, Box::new(JsonStateCreator));
-        Ok(Self { creators })
+    pub fn initialize_for_experiment_run(_config: &PackageInitConfig) -> Result<&'static Self> {
+        static PACKAGE_CREATORS: SyncOnceCell<OutputPackageCreators> = SyncOnceCell::new();
+        PACKAGE_CREATORS.get_or_try_init(|| {
+            tracing::debug!("Outputializing Output Package Creators");
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/package/simulation/output/creator.rs:22]: These are kind of funny but I don't think we should keep them in I'm afraid :( They're just a bit confusing and I could see them actually being distracting while debugging.

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]
One small thing 🎉

File: packages/engine/lib/execution/src/package/simulation/context/creator.rs (hunk 1 above)
Comment (TimDiekmann) [reply packages/engine/lib/execution/src/package/simulation/context/creator.rs]: When you replace `Init` in `Initializing` 🙃 : 927a67ede775f1cc85f0b31f31a90a1a5c6f03be

**Pull request merged**